providers/netty4/src/main/java/org/asynchttpclient/netty/handler/Processor.java

```diff
@@ -71,11 +75,41 @@ public void channelRead(final ChannelHandlerContext ctx, Object msg) throws Exce
                 ac.call();
                 Channels.setDiscard(channel);
             }
+            ReferenceCountUtil.release(msg);
+
 
         } else if (attribute instanceof NettyResponseFuture) {
             NettyResponseFuture<?> future = (NettyResponseFuture<?>) attribute;
             protocol.handle(channel, future, msg);
 
+        } else if (attribute instanceof StreamedResponsePublisher) {
```

Comment:
In this case the stream will simply be notified with the exception via `Subscriber.onError`.

If we did attach the publisher to the future, I *think* we'd have to make a mutable field in `NettyResponseFuture`.  It appeared to me that everything in `NettyResponseFuture` is immutable, and I assumed there was a reason for that - by keeping the only mutable thing the channel attribute, that makes it easier to reason about the state of the system.

On the two crash conditions - the second one is not impacted here.  It's only once we start receiving a body that we change the attribute, before that, all the handling is the same.  For the first crash condition, I think it would be quite straight forward to modify the exception handling such that connections are still retried.  Note that unlike other `AsyncHandlers`, the `StreamedResponsePublisher` changes the state of the pipeline (inserts a handler), and requires events, such as the exception itself, to be forwarded down the channel so that the subscriber that's handling the body receives them.  This is why I think the handling for this should (or must) be in the `Processor`, it requires custom handling at that level.

I'll have a closer look at fixing the exception handling to implement retry - I've realised a `StreamedAsyncHandler` can always return `ABORT` from `onStream` if it's invoked a second time, and that's what we'll do in Play, so I don't think retrying is that big an issue.